lib/suma/api/auth.rb

```diff
@@ -117,71 +47,36 @@ def check_existing_customer!(email, phone, password)
             subject_id: customer.id,
           )
         end
-        # Email may not be verified, but we don't want to send them an email about verification yet.
-        # We can do that later.
-        # customer.add_reset_code({transport: "email"}) unless customer.email_verified?
-        customer.add_reset_code({transport: "sms"}) unless customer.phone_verified?
-        set_customer(customer)
-        create_session(customer)
+        customer.add_reset_code({transport: "sms"})
         status 200
-        present customer, with: Suma::API::CurrentCustomerEntity, env:
+        present({})
       end
     end
-  end
-
-  resource :auth do
-    desc "Log in using phone and password"
-    params do
-      optional :phone, us_phone: true, allow_blank: false
-      optional :email, allow_blank: false
-      exactly_one_of :phone, :email
-      requires :password, type: String, allow_blank: false
-    end
-    post do
-      if current_customer?
-        env["warden"].logout
-        env["warden"].clear_strategies_cache!
-      end
-      customer = authenticate!
-      create_session(customer)
-      status 200
-      present customer, with: Suma::API::CurrentCustomerEntity, env:
-    end
 
-    desc "Verify the current customer phone number using the given token"
     params do
-      requires :token
+      requires :phone, us_phone: true, type: String, coerce_with: NormalizedPhone
+      requires :token, type: String, allow_blank: false
     end
     post :verify do
-      me = current_customer
+      guard_authed!
+      me = Suma::Customer.with_us_phone(params[:phone])
       begin
-        Suma::Customer::ResetCode.use_code_with_token(params[:token]) do |code|
-          invalid!("Invalid verification code") unless code.customer === me
-          code.verify
-          code.customer.save_changes
-          me.refresh
+        Suma::Customer::ResetCode::Unusable if me.nil?
+        unless Suma::Customer.skip_verification?(me)
+          Suma::Customer::ResetCode.use_code_with_token(params[:token]) do |code|
+            raise Suma::Customer::ResetCode::Unusable unless code.customer === me
+          end
         end
       rescue Suma::Customer::ResetCode::Unusable
-        invalid!("Invalid verification code")
+        merror!(401, "Sorry, that token is invalid or the phone number is not in our system.")
       end
 
+      set_customer(me)
+      create_session(me)
       status 200
       present me, with: Suma::API::CurrentCustomerEntity, env:
```

Comment:
Why is there an empty "env" key?

Reply:
`env` is a magical global in the endpoint for most Rack frameworks, like `request` (and it's `env:` as shorthand for `env: env`. It's used by the `CurrentCustomerEntity` (or was, if it's not right now).

Reply:
Okay, I was totally thrown off by the syntax, makes sense. 👍🏼